src/spring-cloud/azext_spring_cloud/tanzu.py

```diff
@@ -191,7 +193,11 @@ def tanzu_app_deploy(cmd, client, resource_group, service, name, artifact_path=N
     4. Query build result from build service
     5. Send build result id to deployment
     '''
-    deployment_name = _assert_deployment_exist_and_retrieve(cmd, client, resource_group, service, name).name
+    deployment = _assert_deployment_exist_and_retrieve(cmd, client, resource_group, service, name)
+    deployment_name = deployment.name
+    # update config-file-patterns for deployment
+    if config_file_patterns:
+        _update_deployment_pattern(client, resource_group, service, name, deployment, config_file_patterns)

```

Comment:
What if the build failed, should we still need to update pattern? If yes, it need to be put right before step 5. Otherwise, I would suggest polling the poller right before step 5 to save the waiting time.